waspc/src/Wasp/Generator.hs

```diff
@@ -18,26 +19,38 @@ import Wasp.Generator.DbGenerator (genDb)
 import qualified Wasp.Generator.DbGenerator as DbGenerator
 import Wasp.Generator.DockerGenerator (genDockerFiles)
 import Wasp.Generator.FileDraft (FileDraft, write)
+import Wasp.Generator.Monad (Generator, GeneratorError, GeneratorWarning, runGenerator)
 import Wasp.Generator.ServerGenerator (genServer)
 import qualified Wasp.Generator.ServerGenerator as ServerGenerator
 import qualified Wasp.Generator.Setup
 import qualified Wasp.Generator.Start
 import Wasp.Generator.WebAppGenerator (generateWebApp)
+import Wasp.Util ((<++>))
 
 -- | Generates web app code from given Wasp and writes it to given destination directory.
 --   If dstDir does not exist yet, it will be created.
+--   It returns a list of any non-fatal warnings encountered, as well as any errors signaling failure.
```

Comment:
Should we explain here what fatal, non-fatal, and failure mean?
We could say something along the lines of: if there are any errors returned, that means that generator failed, and no new code was written. if no errors were returned, this means generator was successful and generated new version of the project, regardless of the warnings returned. Smth like that?

Reply:
Yep, like the idea of cleaning it up in that direction, thanks 👍🏻